text/2020-11-09-chaos-mesh-workflow.md

```diff
@@ -0,0 +1,192 @@
+# Workflow
+
+## Summary
+
+A built-in workflow engine with frontend support is needed for Chaos Mesh.
+
+## Motivation
+
+Both running chaos experiments parallelly or sequentially are important for
+simulating production errors. Chaos Mesh is shipped with a feature-rich chaos
+toolbox. Composing them together can make it much more powerful. With a workflow
+engine, an experiment can turn from single chaos into a series of chaos with
+some external logic (like checking health).
+
+Previously, the `duration` and `scheduler` fields have enabled a very
+fundamental "workflow" feature, so after implementing the workflow engine, these
+fields should be deprecated.
+
+## Detailed design
+
+The specification for a `Workflow` custom resource can be concluded in the
+following example:
+
+```yaml
+spec:
+  cron: "..."
+  entry: serial-chaos
+  templates:
+  - name: networkchaos
+    type: NetworkChaos
+    duration: "30m"
+    selector:
+      labelSelectors:
+        "component": "tikv"
+    delay:
+      latency: "90ms"
+      correlation: "25"
+      jitter: "90ms"
+  - name: iochaos
+    type: IOChaos
+    duration: "10m"
+    selector:
+      labelSelectors:
+        app: "etcd"
+    volumePath: /var/run/etcd
+    path: /var/run/etcd/**/*
+    errno: 5
+    percent: 50
+  - name: parallel-chaos
+    type: Parallel
+    tasks:
+    - ref: networkchaos
+    - ref: iochaos
+  - name: check
+    type: Task
+    task:
+      image: 'alpine:3.6'
+      command: ['/app']
+    branch:
+    - when: "stdout == 'continue'"
+      ref: serial-chaos
+    - when: “stdout == ‘continue’”
```

Comment:
what's the difference between these two "continue"

Reply:
Which means, both of them will execute. They have no difference :smile_cat: